src/darwin/input_hook.c

```diff
@@ -592,15 +593,16 @@ static inline void process_modifier_changed(uint64_t timestamp, CGEventRef event
 			process_key_released(timestamp, event_ref);
 		}
 	}
-	/* FIXME This should produce a modifier mask for the caps lock key!
 	else if (keycode == kVK_CapsLock) {
-		// Process as a key pressed event.
-		process_key_pressed(timestamp, event_ref);
-		
-		// Set the caps-lock flag for release.
-		caps_down = true;
+		if (caps_down) {
```

Comment:
We just need to check the mask above for the state of the mask.  Something like `if (current_modifiers & MASK_CAPS_LOCK) {` should do the trick.